sonic-chassisd/scripts/chassisd

```diff
@@ -566,6 +618,398 @@ class ModuleUpdater(logger.Logger):
+                self.chassis_state_db = daemon_base.db_connect("CHASSIS_STATE_DB")
+
+            # Fetch the dpu_midplane_link_state
+            return self.chassis_state_db.hget(key, "dpu_midplane_link_state")
+
+        except Exception as e:
+            self.log_error(f"Unexpected error: {e}")
+
+    def _convert_to_dict(self, data):
+        """
+        Converts SWIG proxy object or native dict to a Python dictionary.
+        """
+        if isinstance(data, dict):
+            return data  # Already a dict, return as-is
+        else:
+            return dict(data)  # Convert SWIG proxy object to dict
+
+    def _get_current_time_str(self):
+        """Returns the current time as a string in 'YYYY_MM_DD_HH_MM_SS' format."""
+        return datetime.now().strftime("%Y_%m_%d_%H_%M_%S")
+
+    def _get_history_path(self, module, file_name):
+        """Generates the full path for history files."""
+        return os.path.join(REBOOT_CAUSE_DIR, module.lower(), "history", file_name)
+
+    def _is_first_boot(self, module):
+        """Checks if the reboot-cause file indicates a first boot."""
+        file_path = os.path.join(REBOOT_CAUSE_DIR, module.lower(), "reboot-cause.txt")
+
+        try:
+            with open(file_path, 'r') as f:
+                content = f.read().strip()
+                return content == "First boot"
+        except FileNotFoundError:
+                return False
+
+    def persist_dpu_reboot_time(self, module):
+        """Persist the current reboot time to a file."""
+        time_str = self._get_current_time_str()
+        path = self._get_history_path(module, "prev_reboot_time.txt")
+
+        os.makedirs(os.path.dirname(path), exist_ok=True)
+        with open(path, 'w') as f:
+            f.write(time_str)
+
+    def retrieve_dpu_reboot_time(self, module):
+        """Retrieve the persisted reboot time from a file."""
+        path = self._get_history_path(module, "prev_reboot_time.txt")
+
+        try:
+            with open(path, 'r') as f:
+                return f.read().strip()
+        except FileNotFoundError:
+            return None
+
+    def persist_dpu_reboot_cause(self, reboot_cause, module):
+        """Persist the reboot cause information and handle file rotation."""
+        prev_reboot_time = self.retrieve_dpu_reboot_time(module)
+        if prev_reboot_time is None:
+            prev_reboot_time = self._get_current_time_str()
+
+        file_name = f"{prev_reboot_time}_reboot_cause.txt"
+        prev_reboot_path = self._get_history_path(module, "prev_reboot_time.txt")
+
+        if os.path.exists(prev_reboot_path):
+            os.remove(prev_reboot_path)
+
+        file_path = self._get_history_path(module, file_name)
+        try:
+            dt_obj = datetime.strptime(prev_reboot_time, "%Y_%m_%d_%H_%M_%S")
+        except ValueError:
+            dt_obj = datetime.now()
+
+        formatted_time = dt_obj.strftime("%a %b %d %I:%M:%S %p UTC %Y")
+
+        reboot_cause_dict = {
```

Comment:
The `get_reboot_cause` is supposed to return a tuple - a reboot cause and an accompanying description ([from HLD](https://github.com/sonic-net/SONiC/blob/9acaa87252412b3e07d21ec85edcd2c33f2aec3d/doc/smart-switch/pmon/smartswitch-pmon.md?plain=1#L614)) when we perform `json.dump` with a tuple, it is saved as a list can we split here as "cause" and "description" separately? otherwise the `# Publish the reboot cause information to CHASSIS_STATE_DB` step fails in chassisd because we are performing `hset` to chassis_state_db with a list

Reply:
@gpunathilell Done

Reply:
@gpunathilell Done

Reply:
Resolved